### PR TITLE
feat: show Terraform error details

### LINKED ISSUE #6453: Terraform errors don't give line numbers, making it hard to debug
Error from Coder:

```sh
⧗  Detecting persistent resources 
  Terraform 1.1.9
  Error: Unsupported attribute
  This object has no argument, nested block, or exported attribute named "namespace".
✔ Detecting persistent resources [2579ms]
✘ Cleaning Up [9ms]
template import provision for start: recv import provision: plan terraform: terraform plan: exit status 1
Run 'coder templates push --help' for usage.        
```

Error from `terraform apply`

```sh
╷
│ Error: Unsupported attribute
│ 
│   on main.tf line 21, in locals:
│   21:   namespace = "coder-ws-${lower(data.coder_workspace.me.namespace)}"
│ 
│ This object has no argument, nested block, or exported attribute named "namespace".
╵
```

### PR DESCRIPTION
**Pull request by mtojek**
Resolves: https://github.com/coder/coder/issues/6453

This PR enables Terraform diagnostics details in the site view and CLI. Unfortunately, the original formatter is in the `internal` package, so I had to implement a similar one.

Screenshot 2023-03-17 at 12 43 12

Screenshot 2023-03-17 at 12 42 10

### CODE REVIEW

File: provisioner/terraform/executor.go [hunk 1]
```diff
@@ -509,7 +512,7 @@ func convertTerraformLogLevel(logLevel string, sink logSink) proto.LogLevel {
 		return proto.LogLevel_DEBUG
 	case "info":
 		return proto.LogLevel_INFO
-	case "warn":
+	case "warn", "warning":
```
Comment (mafredri) [inline provisioner/terraform/executor.go:515]: Why do we need both?

File: provisioner/terraform/diagnostic.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+			copy(values, snippet.Values)
+			sort.Slice(values, func(i, j int) bool {
+				return values[i].Traversal < values[j].Traversal
+			})
+
+			_, _ = buf.WriteString("    ├────────────────\n")
+			for _, value := range values {
+				_, _ = fmt.Fprintf(buf, "    │ %s %s\n", value.Traversal, value.Statement)
+			}
+		}
```
Comment (mafredri) [inline provisioner/terraform/diagnostic.go:65]: It seemed like the original (internal) code is slightly more complex, do we lose any details in error output or are they surfaced some other way? It looked like function calls/args were handled in internal, but not here.

**Review by mafredri — APPROVED** [review]
Looks good in general, but I couldn't really see how/if the existing internal formatting is used at all any longer, in which case we may lose error detail?

**Review by bpmct — APPROVED** [review]

File: provisioner/terraform/diagnostic.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	"sort"
+	"strings"
+
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+// This implementation bases on the original Terraform formatter, which unfortunately is internal:
+// https://github.com/hashicorp/terraform/blob/6b35927cf0988262739a5f0acea4790ae58a16d3/internal/command/format/diagnostic.go#L125
+
+func FormatDiagnostic(diag *tfjson.Diagnostic) []string {
```
[the remark this answers is not in the document]
Comment (mtojek) [reply provisioner/terraform/diagnostic.go]: Good call

**Pull request merged**